parse('')" cannot be evaluated. version string portion was too short or too long

Hello guys.

We are facing some bug since months but it's not always happening.
So that's why we didn't report it, but recently I do believe we have to report the issue because it's happening more.
When we are building, we are getting:

C:\Users\Owner\.nuget\packages\gitinfo\3.0.4\build\GitInfo.targets(226,12): error MSB4184: The expression "[System.Version]::Parse('')" cannot be evaluated. Version string portion was too short or too long.

We have to retry building few times and then it's working fine.

Is there any solution to fix this issue ?

Thank you so much.

Best regards

--- a/src/GitInfo/build/GitInfo.targets
+++ b/src/GitInfo/build/GitInfo.targets
@@ -223,8 +223,12 @@
       <_GitCurrentVersion>$([System.Text.RegularExpressions.Regex]::Match("$(_GitOutput)", "\d+\.\d+\.\d+").Value)</_GitCurrentVersion>
     </PropertyGroup>
 
-    <Error Condition="$([System.Version]::Parse('$(_GitCurrentVersion)').CompareTo($([System.Version]::Parse('$(GitMinVersion)')))) &lt; 0"
-			   Text="Required minimum git version is $(GitMinVersion) but found $(_GitCurrentVersion)." />
+    <Warning Condition="$(_GitCurrentVersion) == ''"
+             Code="GI007"
+             Text="Could not determine git version from output '$(_GitOutput)'. Required minimum git version is $(GitMinVersion)." />
+
+    <Error Condition="$(_GitCurrentVersion) != '' and $([System.Version]::Parse('$(_GitCurrentVersion)').CompareTo($([System.Version]::Parse('$(GitMinVersion)')))) &lt; 0"
+			     Text="Required minimum git version is $(GitMinVersion) but found $(_GitCurrentVersion)." />
   </Target>
 
   <Target Name="_GitRoot" Returns="$(GitRoot)" Condition="'$(GitRoot)' == ''">

